web/src/containers/P2P/TradesHistory/index.tsx

```diff
@@ -83,35 +112,8 @@ const TradesHistory: FC = (): ReactElement => {
                     />}
                 </div>
             </div>
-        ];
-    };
-
-    const retrieveData = () => {
-        return [...list].map(item => renderTableRow(item));
-    };
-
-    const onRowClick = useCallback((index: string) => {
-        const orderId = list[index]?.id;
-        orderId && history.push(`/p2p/order/${orderId}`);
-    }, [history, list]);
-
-    const renderTableRow = (item: P2POrder) => {
-        const { created_at, side, amount, state, user_uid } = item;
-        const { price, quote, base, user } = item.offer;
-        const sideColored = setTradesType(side);
-        const stateColored = setStateType(state);
-        const pricePrecision = currencies.find(item => item.id === quote)?.precision;
-        const amountPrecision = currencies.find(item => item.id === base)?.precision;
-
-        return [
-            localeDate(created_at, 'fullDate'),
-            <span style={{ color: sideColored.color }}>{sideColored.text}</span>,
-            `${Decimal.format(price, pricePrecision || DEFAULT_FIAT_PRECISION, ',')} ${base.toUpperCase()}/${quote.toUpperCase()}`,
-            `${Decimal.format(amount, amountPrecision || DEFAULT_CCY_PRECISION, ',')} ${base.toUpperCase()}`,
-            <span>{user?.user_nickname || user_uid}</span>,
-            <span style={{ color: stateColored.color }}>{stateColored.text}</span>,
-        ];
-    };
+        );
+    }, [list, firstElemIndex, lastElemIndex, total, page, nextPageExists, onClickPrevPage, onClickNextPage, renderHeaders, retrieveData, onRowClick, translate]);
```

Comment:
Please check all these values here
I'm sure some of them shouldn't be here

Reply:
Fixed, thanks.